ddtrace/appsec/_iast/_stacktrace.c

```diff
@@ -39,68 +51,82 @@
  * @return Tuple, string and integer.
  **/
 static PyObject*
-get_file_and_line(PyObject* Py_UNUSED(module), PyObject* args)
+get_file_and_line(PyObject* Py_UNUSED(module), PyObject* cwd_obj)
 {
-    PyThreadState* tstate = PyThreadState_GET();
-    PyFrameObject* frame;
-    PyObject* filename_o;
+    PyThreadState* tstate = PyThreadState_Get();
+    if(!tstate) {
+        return NULL;
+    }
+
     int line;
+    PyObject* filename_o = NULL;
+    PyObject* result = NULL;
+    PyObject *cwd_bytes = NULL;
+    char* cwd = NULL;
 
-    PyObject *cwd_obj = Py_None, *cwd_bytes;
-    char* cwd;
-    if (!PyArg_ParseTuple(args, "O", &cwd_obj))
+    if (!PyUnicode_FSConverter(cwd_obj, &cwd_bytes)) {
```

Comment:
Change this if you want to return `None` for some reason in casee you pass something invalid. Right now it shows:

```
>>> import _stacktrace as s
>>> s.get_info_frame(34.4)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: expected str, bytes or os.PathLike object, not float
```